Javatrix.java

```diff
@@ -22,6 +22,37 @@
     private int m;
     private int n;
 
+     /**
+     * Constructor 1D array.
+     * @param m number of rows
+     * @param vals stored by columns
+     */
+    public Javatrix(double[] vals, int m)throws IllegalArgumentException
+    {
+        int rows = m;
+        int columns = (vals.length) / m;
+        int reman = (vals.length) % m;
+
+        if (reman != 0)
+        {
+            String ex = "Array length must me a multiple of m.";
+            throw new IllegalArgumentException(ex);
+        }
+        else 
```

Comment:
Don't need an else here - when the if test passes, the exception will be thrown and the program exited.